### ISSUE TITLE
Expose `ttx` as a module

### ISSUE THREAD

**Issue by mattlag** [web-alchemy/fonttools#4]
If I am missing something, please let me know. I am trying to use this as a module in a nodejs project.

It seems as if only `subset` and `instantiateVariableFont` are available. I would love to be able to add `ttx` to this, like:

```
const {
  subset,
  instantiateVariableFont,
  ttx
} = require('@web-alchemy/fonttools')
```

then be able to call `ttx(xmlFile)` or `ttx(otfFile)` to do XML to OTF or OTF to XML conversions.

**Comment by monochromer** [comment]
Hi. At the moment `ttx` available only as CLI:
```shell
npx -p @web-alchemy/fonttools ttx <params>
```

I will try to add functionality within a couple of weeks.

**Comment by mattlag** [comment]
Thank you, that would be amazing!

**Comment by monochromer** [comment]
@mattlag you can test the feature installing package from git branch:

```shell
npm add https://github.com/web-alchemy/fonttools.git#feature/ttx-module
```

Example of converting binary files to xml:

```javascript
const fs = require('node:fs');
const { ttx } = require('@web-alchemy/fonttools');

(async () => {
  const outputTtxBuffer = await ttx('./font.ttf'); // also accept `URL` and `Buffer`
  await fs.promises.writeFile('./font.ttx', outputTtxBuffer);
  
  const outputOtxBuffer = await ttx('./font.otf');
  await fs.promises.writeFile('./font.otx', outputOtxBuffer);
})();
```

Example of converting xml files to binary files:

```javascript
const fs = require('node:fs');
const { ttx } = require('@web-alchemy/fonttools');

(async () => {
  const ttxBuffer = await ttx('./font.ttx');
  await fs.promises.writeFile('./font.ttf', ttxBuffer);
  
  const otxBuffer = await ttx('./font.otx');
  await fs.promises.writeFile('./font.otf', otxBuffer);
})();
```

Example of converting xml files to binary files with encoding to `woff2`:

```javascript
const fs = require('node:fs');
const { ttx } = require('@web-alchemy/fonttools');

(async () => {
  const ttxBuffer = await ttx('./font.ttx', [
    ['--flavor', 'woff2']
  ]);
  await fs.promises.writeFile('./font.ttf', ttxBuffer);
  
  const otxBuffer = await ttx('./font.otx', [
    ['--flavor', 'woff2']
  ]);
  await fs.promises.writeFile('./font.otf', otxBuffer);
})();
```

If it's ok, i will be merge this feature and publish new version of package.

**Comment by mattlag** [comment]
This is fantastic! Thank you so much!

I was able to test many cases:
- ttx > ttf
- ttf > ttx
- otf > ttx
- otx > otf

( I actually didn't know about the otx - otf relationship, this is great!)

I was not able to get the `['--flavor', 'woff2']` to work... seemed to hang on reading the `htmx` table. But, i'm not interested in woff2, so that is fine for me. I tried passing it both a ttx file and an otx file. All I got was this:

![image](https://github.com/web-alchemy/fonttools/assets/5897669/42827c2a-b226-4ed3-b27d-e54c79481fba)


But, i would say, go ahead and integrate it!
